app/routers/google_connect.py

```diff
@@ -0,0 +1,228 @@
+from datetime import datetime
+from fastapi import Depends, APIRouter
+from starlette.responses import RedirectResponse
+
+from google.auth.transport.requests import Request as google_request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+
+from app.database.models import Event, User, OAuthCredentials, UserEvent
+from app.database.database import get_db, SessionLocal
+from app.routers.profile import router as profile_router
+from app.config import CLIENT_SECRET_FILE
+from app.routers.event import create_event
+from loguru import logger
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+router = APIRouter(
+    prefix="/google",
+    tags=["sync"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/sync")
+async def google_sync(session=Depends(get_db)) -> RedirectResponse:
+    '''Sync with Google - if user never synced with google this funcion will take
+    the user to a consent screen to use his google calendar data with the app.
+    '''
+
+    user = get_active_user(session)
+    credentials, status = get_credentials_from_db(user)
+
+    if status:
```

Comment:
About which line you're talking about?

Reply:
???

Reply:
The whole `if` block.

Reply:
It's must be there or otherwise nothing would work as expected.

Reply:
> It's must be there or otherwise nothing would work as expected.

Why do you think so?

> and it's already splited into functions.
> it only have to checks that one condition.
 
Functions should usually be 3-5 lines long.
This function does more than one thing - it's hard to read and it's hard to test.
Please split it.

Reply:
Did it! :)
it's much better you're right 💯 %.